src/prototype_creep.js

```diff
@@ -344,19 +344,23 @@ Creep.prototype.killPrevious = function(path) {
     return false;
   }
 
+  let killWho;
+  let killWhoName;
   if (this.ticksToLive < previous.ticksToLive) {
-    if (this.ticksToLive > 100) {
-      this.log('kill me: me: ' + this.ticksToLive + ' they: ' + previous.ticksToLive);
-    }
-    this.memory.killed = true;
-    this.suicide();
+    killWhoName = 'me';
+    killWho = this;
   } else {
-    if (previous.ticksToLive > 100) {
-      this.log('kill other: me: ' + this.ticksToLive + ' they: ' + previous.ticksToLive);
-    }
-    previous.memory.killed = true;
-    previous.suicide();
+    killWhoName = 'other';
+    killWho = previous;
+  }
+
+  if (killWho.ticksToLive > killWho.memory.timeToTravel) {
```

Comment:
nice cleanup, but since when do we have a `creep.memory.timeToTravel`?

Reply:
We have it for a long time, this indicates when the next creep need to be spawned e.g. for reserver, upgrader or sourcerer